pkg/partmgr/partmgr.go

```diff
@@ -157,8 +161,19 @@ func (p *PartitionManager) createNewPartition(t int64) (*DBPartition, error) {
 		return nil, err
 	}
 	p.currentPartitionInterval = partition.partitionInterval
```

Comment:
It would be better to separate creation logic from update current partition details